.github/workflows/dev-test.yml

```diff
@@ -0,0 +1,140 @@
+
+    - uses: actions/setup-node@v1.4.2
+      with:
+        node-version: '14.x' # optional, default is 10.x
+
+    - name: Install packages
+      run: npm ci
+      working-directory: gitgitgadget
+
+    - name: Set git smtp config
+      shell: bash
+      env:
+        # see ci-helper.test.ts for info on setting these
+        SMTPUSER: ${{ secrets.SMTPUSER }}
+        SMTPPASS: ${{ secrets.SMTPPASS }}
+        SMTPHOST: ${{ secrets.SMTPHOST }}
+        SMTPOPTS: ${{ secrets.SMTPOPTS }}
+      run: |
+        git config --add gitgitgadget.cismtpuser $SMTPUSER
+        git config --add gitgitgadget.cismtphost $SMTPHOST
+        git config --add gitgitgadget.cismtpopts "$SMTPOPTS"
+        git config --add gitgitgadget.cismtppass $SMTPPASS
+      if: env.SMTPUSER
+      working-directory: gitgitgadget
+
+    - name: Set git test repo config
+      shell: bash
+      run: |
+        git config --add gitgitgadget.githubtest.githubuser $GITHUB_REPOSITORY_OWNER
+        git config --add gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubrepo $GGG_REPOSITORY
+        git config --add gitgitgadget.$GITHUB_REPOSITORY_OWNER.githubtoken $GGG_TOKEN
+      if: env.GGG_TOKEN
+      working-directory: gitgitgadget
+
+    - name: Run build
+      run: npm run build
+      working-directory: gitgitgadget
+
+    - name: Run tests
+      run: npm run test
+      working-directory: gitgitgadget
+
+  build-win:
+    runs-on: windows-latest
+
+    # job level env vars used for a couple of steps
+    env:
+      # see github-glue.test.ts for info on setting these
+      GGG_TOKEN: ${{ secrets.GGG_TOKEN }}
+      GGG_REPOSITORY: ${{ secrets.GGG_REPOSITORY }}
+
+    steps:
+    # Check out repo under sub-dir so other repo can be checked out
+    - uses: actions/checkout@v2
+      with:
+        path: gitgitgadget
+
+    # Check out github-glue.test.ts repo if configured for it
+    - uses: actions/checkout@v2
+      with:
+        repository: ${{ github.repository_owner }}/${{ env.GGG_REPOSITORY }}
+        token: ${{ env.GGG_TOKEN }}
+        path: ${{ env.GGG_REPOSITORY }}
+      if: env.GGG_TOKEN
+
+    - uses: actions/setup-node@v1.4.2
+      with:
+        node-version: '14.x' # optional, default is 10.x
+
+    - name: Install packages
+      run: npm ci
+      working-directory: gitgitgadget
+
+    - name: Set git smtp config
+      shell: cmd
+      env:
+        # see ci-helper.test.ts for info on setting these
+        SMTPUSER: ${{ secrets.SMTPUSER }}
+        SMTPPASS: ${{ secrets.SMTPPASS }}
+        SMTPHOST: ${{ secrets.SMTPHOST }}
+        SMTPOPTS: ${{ secrets.SMTPOPTS }}
+      run: |
+        git config --add gitgitgadget.cismtpuser %SMTPUSER%
+        git config --add gitgitgadget.cismtphost %SMTPHOST%
+        git config --add gitgitgadget.cismtpopts "%SMTPOPTS%"
+        git config --add gitgitgadget.cismtppass %SMTPPASS%
+      if: env.SMTPUSER
+      working-directory: gitgitgadget
+
+    - name: Set git test repo config
+      shell: cmd
```

Comment:
How about using `bash` here, and then turning it into a [matrix workflow](https://www.edwardthomson.com/blog/github_actions_6_fail_fast_matrix_workflows.html)?

Reply:
Sure.  Not sure why I didn't.